.github/workflows/build.yaml

```diff
@@ -35,6 +35,41 @@ jobs:
         with:
           node-version: "16"
 
+      - name: Fetch dependencies from cache
+        id: cache-node-modules
+        uses: actions/cache@v3
+        with:
+          path: "node_modules"
```

Comment:
✍🏼 Since these jobs like `fmt` don't require any dependencies besides the root ones, I left the caching to only these `node_modules`. Happy to change back if desired.

Reply:
I think if we do this we should make the postinstall script not install the extra dependencies in `test` otherwise it will do that every time since they will not be cached.

Reply:
I am not sure but it might be a little less efficient overall to do it this way because we would have one cache with node_modules, another cache with node_modules plus test/node_modules.

Reply:
Also if the cache contents are different I am not sure they should share the same fallback restore keys.  Sometimes it would have node_modules, sometimes it would also include test/node_modules.

Reply:
Ahhh okay. Good points. I think I'll change back to how it was before. I do want to revisit the caching in a follow-up PR because `setup-node` supports caching out of the box, so no need to do it manually. When I do that, we can revisit the idea of splitting up caches.